zebra-state/src/service/finalized_state/disk_format/transparent.rs

```diff
@@ -504,8 +504,8 @@ fn address_variant(address: &transparent::Address) -> u8 {
     match (address.network(), address) {
         (Mainnet, PayToPublicKeyHash { .. }) => 0,
         (Mainnet, PayToScriptHash { .. }) => 1,
-        (Testnet, PayToPublicKeyHash { .. }) => 2,
-        (Testnet, PayToScriptHash { .. }) => 3,
+        (Testnet(_), PayToPublicKeyHash { .. }) => 2,
+        (Testnet(_), PayToScriptHash { .. }) => 3,
```

Comment:
This might be incorrect for custom testnets if they change their address formats. What does `zcashd` do here?

Reply:
> What does zcashd do here?

I'm not sure, but it looks like zcashd doesn't store the prefix. (though there are prefixes defined in the network/chain parameters)